### PR TITLE
Feat/#22 set layout

### PR DESCRIPTION
**Pull request by Poukam-Pierre**
Setting up layout for admin project.

### CODE REVIEW

File: apps/admin-ui/components/Layout/LanguageSwapper.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { Icon } from "@iconify/react";
+
+type CorrespondingShortLanguage = Record<string, string>;
+
+export default function LanguageSwapper() {
+    const correspondingShortLanguage: CorrespondingShortLanguage = {
+        'fr': 'Français',
+        'en': 'Anglais'
+    }
+    const supportedLanguages: string[] = ['Français', 'Anglais']
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/LanguageSwapper.tsx:13]: change the `supportedLanguages` to an `enum`

```ts
enum supportedLanguages = {
english = 'English',
french = 'Français'
}
```

and use it like 

```ts
Object.values(supportedLanguages)
```
which will give you an array of the values that  you need.
this approach is clean but permits for extensibility in a clean way

File: apps/admin-ui/components/Layout/LanguageSwapper.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import languageIcon from '@iconify-icons/material-symbols/language'
+import { Icon } from "@iconify/react";
+
+type CorrespondingShortLanguage = Record<string, string>;
+
+export default function LanguageSwapper() {
+    const correspondingShortLanguage: CorrespondingShortLanguage = {
+        'fr': 'Français',
+        'en': 'Anglais'
+    }
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/LanguageSwapper.tsx:12]: the languages should present in the required format.
```suggestion
    const correspondingShortLanguage: CorrespondingShortLanguage = {
        'fr': 'Français',
        'en': 'English'
    }
```

this permits english to always display like that and read and understood by all.

File: apps/admin-ui/components/Layout/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                '/masses/create': 'Create',
+                '/masses/modify': 'Modify',
+                '/masses/delete': 'Cancel',
+            }
+        },
+        {
+            title: 'Finances',
+            links: {
+                '/finances': 'Finances',
+                'finances/widrawal': 'Widrawal',
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Header.tsx:33]:
```suggestion
                '/finances/withdrawal': 'Withdrawal',
```

File: apps/admin-ui/components/Layout/Header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            }
+        },
+    ]
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Header.tsx:36]: if the application intends to translate text, then even the breadcrumbs should be translated. 
this means, the `title` and `link` values should be translated

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import { Box, Breadcrumbs, Link, Typography } from "@mui/material";
+import { BreadcrumbsNameMaps } from "./Header";
+import { useRouter } from "next/router";
+
+interface BreadcrumbProps {
+    breadcrumbs: BreadcrumbsNameMaps
+}
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx:7]: no need to create this `props` interface with only one value in it.
send that value directly.

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+export default function Breadcrumb({ breadcrumbs: {
+    title,
+    links
+} }: BreadcrumbProps) {
+    const { pathname } = useRouter()
+    const pathnames = pathname.split('/').filter((x) => x)
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx:14]: what is the essence of the filter at the end?
from what i see, it only takes a value and return it, is that not a redundant operation?

also, you can't call this variable ~pathnames~, you can't split `a` pathname and have multiple pathnames.
it's the `subPaths` or simply `paths`.

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    console.log(pathname.split('/'), pathnames)
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx]: get rid of this

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                <Link
+                    variant="body2"
+                    underline="hover"
+                    color="inherit"
+                >
+                    Management
+                </Link>
+                {pathnames.map((_, index) => {
+                    const last = index === pathnames.length - 1;
+                    const to = `/${pathnames.slice(0, index + 1).join('/')}`;
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx:39]: variable names are not self explanatory.

`isLastPath` and `pathRoute` will be better names, than just ~last~ and ~to~ from these names, we can't infer their meaning or usage

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                    padding: 0
+                }}
+            >{title}</Typography>
+            <Breadcrumbs>
+                <Link
+                    variant="body2"
+                    underline="hover"
+                    color="inherit"
+                >
+                    Management
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx:35]: this text should be translated
and I'm not sure it'll appear in all routes,

think about it and see if it should not be gotten rid of

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+                            href={to}
+                        >
+                            {links[to]}
+                        </Link>
+                    )
+                })}
+            </Breadcrumbs>
+        </Box>
+    );
+}
```
Comment (Tchakoumi) [inline apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx]:
```suggestion
import { Box, Breadcrumbs, Link, Typography } from "@mui/material";
import { BreadcrumbsNameMaps } from "./Header";
import { useRouter } from "next/router";

export default function Breadcrumb({ title, links }: BreadcrumbsNameMaps) {
  const { pathname } = useRouter();
  //The filter is to remove the trailing `/` that exists in the pathname variable.
  const paths = pathname.split("/").filter((x) => x !== "/");
  return (
    <Box
      sx={{
        display: "grid",
        textAlign: "start",
      }}
    >
      <Typography
        variant="h1"
        sx={{
          fontSize: "36px",
          lineHeight: "44px",
          letterSpacing: "-0.02px",
          padding: 0,
        }}
      >
        {title}
      </Typography>
      <Breadcrumbs>
        <Link variant="body2" underline="hover" color="inherit">
          Management
        </Link>
        {paths.map((_, index) => {
          const isPathLast = index === paths.length - 1;
          const pathRoute = `/${paths.slice(0, index + 1).join("/")}`;
          return isPathLast ? (
            <Typography key={index} color="primary">
              {links[pathRoute]}
            </Typography>
          ) : (
            <Link
              key={index}
              variant="body2"
              underline="hover"
              color="inherit"
              href={pathRoute}
            >
              {links[pathRoute]}
            </Link>
          );
        })}
      </Breadcrumbs>
    </Box>
  );
}
```

File: apps/easy-messe/components/OfferMass/letOfferMass.schema.js
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/letOfferMass.schema.js:1]: if this schema is not used elsewhere, then it should be taken to the spot where the form is declared, and the errors translated with `formatMessage` at once.

File: apps/easy-messe/components/OfferMass/OfferMassCard.tsx [hunk 11]
```diff
@@ -15,7 +15,7 @@ export default function OfferMassCart({ massInfos }: OfferMassCartProps): JSX.El
     const { formatMessage, formatNumber } = useIntl()
 
 
-    const arrayStaticData: string[] = ['city', 'parish', 'dateTime', 'applicant']
+    const staticDatas: string[] = ['city', 'parish', 'dateTime', 'applicant']
```
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/OfferMassCard.tsx:18]: data does not have plural.
```suggestion
    const staticData: string[] = ['city', 'parish', 'dateTime', 'applicant']
```

**Review by Tchakoumi — CHANGES_REQUESTED** [review]

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx (hunk 6 above)
Comment (Poukam-Pierre) [reply apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx:14]: The filter JS function here is used to remove the empty string which appeared when split.
Actually, I expected the same from the filter function, like taking a value and returning it, so it wasn't useful, but when split, a string appeared empty in the array. So we've added a filter to perform an array without an empty string value.

File: apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx (hunk 9 above)
Comment (Poukam-Pierre) [reply apps/admin-ui/components/Layout/Header/Breadcrumbs.tsx:35]: Okey!

**Review by Tchakoumi — APPROVED** [review]

**Pull request merged**